manifests/service.pp

```diff
@@ -3,16 +3,39 @@
 # Manage PAM file for a specifc service
 #
 define pam::service (
+  $ensure         = 'present',
   $pam_config_dir = '/etc/pam.d',
   $content        = undef,
+  $lines          = undef
 ) {
 
   include pam
 
+  validate_re($ensure, ['^present$', '^absent$'] )
```

Comment:
yes, though bare strings are being depreciated with the Future Parser and sooner or later will start throwing syntax validation warnings.